### PR TITLE
CA-813 Add an integration test that reads Google Service Account Credentials.

### PR DESCRIPTION
**Pull request by wchamber**
Set up the github action to render the service account file from Vault and upload gradle test reports.
Add tasks to build.gradle to allow unit and integration tests to run separately.
Add a  integration test to be exercised (adding gradle dependencies as needed).
Add a stub section to the README about running integration tests locally.

I verified locally that the tests fail when they should.

### CODE REVIEW

File: src/test/java/bio/terra/cloudres/HelloTest.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package bio.terra.cloudres;
```
Comment (yonghaoy) [inline src/test/java/bio/terra/cloudres/HelloTest.java:1]: I deleted the Hello class in my another PR, this is not needed?

File: src/test/java/bio/terra/cloudres/common/IntegrationCredentials.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package bio.terra.cloudres.common;
```
Comment (yonghaoy) [inline src/test/java/bio/terra/cloudres/common/IntegrationCredentials.java:1]: does this should be under testing/ directory?

File: src/test/java/bio/terra/cloudres/common/IntegrationCredentialsTest.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package bio.terra.cloudres.common;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+@Tag("integration")
+public class IntegrationCredentialsTest {
+  @Test
```
Comment (yonghaoy) [inline src/test/java/bio/terra/cloudres/common/IntegrationCredentialsTest.java:8]: I thought the general practice is let the test throws Exception, but we can add
try {
      
   } catch(yourException e) {
      fail("Should not have thrown any exception");
   }

File: .github/workflows/build-and-test.yaml [hunk 4]
```diff
@@ -8,14 +8,38 @@ on:
...
+          echo ::set-output name=vault-token::$VAULT_TOKEN
+          echo ::add-mask::$VAULT_TOKEN
+      - name: Get service account credentials from Vault
+        id: vault-secret-step
+        run: |
+          docker run --rm --cap-add IPC_LOCK \
+            -e "VAULT_TOKEN=${{ steps.vault-token-step.outputs.vault-token }}" \
+            -e "VAULT_ADDR=${VAULT_ADDR}" \
+            vault:1.1.0 \
+            vault read -format json secret/dsde/terra/crl-test/default/service-account.json \
```
[the remark this answers is not in the document]
Comment (wchamber) [reply .github/workflows/build-and-test.yaml]: good idea, thanks.

File: src/test/java/bio/terra/cloudres/HelloTest.java (hunk 1 above)
Comment (wchamber) [reply src/test/java/bio/terra/cloudres/HelloTest.java]: Yep, thanks. just needed a unit test to run.

File: src/test/java/bio/terra/cloudres/common/IntegrationCredentials.java (hunk 2 above)
Comment (wchamber) [reply src/test/java/bio/terra/cloudres/common/IntegrationCredentials.java]: done.

File: src/test/java/bio/terra/cloudres/common/IntegrationCredentialsTest.java (hunk 3 above)
Comment (wchamber) [reply src/test/java/bio/terra/cloudres/common/IntegrationCredentialsTest.java]: I like that that makes it more explicit, done. Do you have a source? Not sure I've seen that before.

Comment (yonghaoy) [reply src/test/java/bio/terra/cloudres/common/IntegrationCredentialsTest.java]: Forgot where I saw the documented source, that might come from some someone else's recommendation during code review before.

Just found an google internal  discussion: https://yaqs.corp.google.com/eng/q/5403980279709696#a6708475597029376

**Pull request merged**